src/benchmarks/micro/libraries/System.Numerics.Tensors/Perf_SignedIntegerTensorPrimitives.cs

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+using System.Linq;
+
+namespace System.Numerics.Tensors.Tests
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.SIMD, Categories.JIT)]
+    [GenericTypeArguments(typeof(sbyte))]
+    [GenericTypeArguments(typeof(short))]
+    [GenericTypeArguments(typeof(int))]
+    [GenericTypeArguments(typeof(long))]
+    public class Perf_SignedIntegerTensorPrimitives<T>
+        where T : unmanaged, IBinaryInteger<T>, IMinMaxValue<T>
+    {
+        [Params(3, 64 + 7, 128 + 7, 256 + 7, 512 + 7, 3 * 512, 6 * 512)]
+        public int BufferLength;
```

Comment:
@adamsitnik and @DrewScoggins are likely going to comment the same thing....

But, this is going to add a significant amount of overhead to the perf runs and likely nearly double their total execution time (the same concern came up for just covering the `Vector64/128/256/512` primitive types, which are doing far less work)

Reply:
Would it make sense to also reduce the set of types being tested to, say, just `int` and `long`? What about removing benchmarking for APIs we feel are less critical?

Reply:
> Would it make sense to also reduce the set of types being tested to, say, just int and long?

I suggest we have float, double, and int, and if there's capacity, also Half.

Reply:
> What about removing benchmarking for APIs we feel are less critical?

Yes. I noted in a separate comment, but I'd make it a little more open-box testing and pick APIs based on having a unique workhorse implementation behind them (most of the operators share an implementation that then just delegates to the operator's op-specific code).

Reply:
> I suggest we have float, double, and int, and if there's capacity, also Half.

This particular test suite only deals with binary integers, so by the sound of it we should trim it down to just `int`?

Reply:
I pushed changes bringing down the total number of benchmarks down to 30. Note that this doesn't cover the entire API surface of TP, just the `BinaryInteger<T>` APIs.